File: problems/array/java/ReverseAnArray.java
import java.util.*;

public class ReverseAnArray {
    
    static int[] reveseArray(int[] array) {
        int iteration = (int) Math.ceil(array.length / 2);

        for( int i=0; i<iteration; i++) {
            int temp = array[i];
            array[i] = array[array.length - 1 - i];
            array[array.length - 1 - i] = temp;
        }

        return array;
    }
    public static void main(String[] args) {
        System.out.println(Arrays.toString(reveseArray(new int[]{ 1, 2, 3, 4, 5 })));
        System.out.println(Arrays.toString(reveseArray(new int[]{ 1, 2, 3, 4, 5, 6 })));
    }
}

File: problems/array/typescript/ReverseAnArray.ts
/*
The problem is that we want to reverse a T[] array in O(N) linear time complexity and we want the algorithm to be in-place as well!
For example: input is [1,2,3,4,5] then the output is [5,4,3,2,1]
*/

function reverseAnArray(array: number[]) {
    const ITERATION = Math.ceil(array.length / 2);
    console.log(ITERATION);
    for (let i = 0; i < ITERATION;  i++) {
        const TEMP = array[i];
        array[i] = array[array.length - 1 - i];
        array[array.length - 1 - i] = TEMP;
    }
    return array;
}

console.log(reverseAnArray([1, 2, 3, 4, 5]));
console.log(reverseAnArray([1, 2, 3, 4, 5, 6]));
